src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -164,7 +167,21 @@ internal bool OnEdit(DocumentId documentId)
                 switch (projectReason)
                 {
                     case ProjectReadOnlyReason.MetadataNotAvailable:
-                        message = "Changes are not allowed if the project wasn't built when debugging started.";
+                        // TODO: Remove once https://github.com/dotnet/roslyn/issues/16657 is addressed
+                        bool deferredLoad = (_vsProject.ServiceProvider.GetService(typeof(SVsSolution)) as IVsSolution7)?.IsSolutionLoadDeferred() == true;
+                        if (deferredLoad)
+                        {
+                            message = "Changes are not allowed if the project wasn't loaded and built when debugging started." + Environment.NewLine + 
```

Comment:
nit: "built and loaded" rather than "loaded and built" (since presumably, you build before loading chronologically)